src/opencl/axcrypt_kernel.cl

```diff
@@ -0,0 +1,115 @@
+/*
+ * This software is Copyright (c) 2018 Dhiru Kholia, Copyright (c) 2016
+ * Fist0urs <eddy.maaalou at gmail.com>, Copyright (c) 2018 magnum, and it is
+ * hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+#include "opencl_device_info.h"
+#include "opencl_misc.h"
+#include "opencl_sha1_ctx.h"
+#include "opencl_aes.h"
+
+#ifndef PLAINTEXT_LENGTH
+#error PLAINTEXT_LENGTH must be defined
+#endif
+
+// input
+typedef struct {
+	uint length;
+	uchar v[PLAINTEXT_LENGTH];
+} axcrypt_password;
+
+// input
+typedef struct {
+	uint version;
+        uint key_wrapping_rounds;
+        uint keyfile_length;
+        uchar salt[16];
+        uchar wrappedkey[24];
+        uchar keyfile[4096];
+} axcrypt_salt;
+
+// output
+typedef struct {
+	uint cracked;
+} axcrypt_out;
+
+#define PUT_64BITS_XOR_MSB(cp, value) ( \
+                (cp)[0] ^= (unsigned char)((value)), \
+                (cp)[1] ^= (unsigned char)((value) >> 8), \
+                (cp)[2] ^= (unsigned char)((value) >> 16), \
+                (cp)[3] ^= (unsigned char)((value) >> 24 ) )
+
+inline int axcrypt_decrypt(__global const axcrypt_password *inbuffer, uint gid, __constant axcrypt_salt *cur_salt, __global axcrypt_out *output)
+{
+	uchar password[PLAINTEXT_LENGTH];
+	uchar keyfile[4096];
+	int password_length = inbuffer[gid].length;
+	uchar salt[16];
+	unsigned char KEK[20], lsb[24], cipher[16];
```

Comment:
Why is lsb 24 bytes when you only ever touch the first 16?

Reply:
Ack. Fixing.